Examples/PcapSplitter/ConnectionSplitters.h

```diff
@@ -145,4 +145,113 @@ class FiveTupleSplitter : public ValueBasedSplitter
 
 		return m_FlowTable[hash];
 	}
+
+	/**
+	 * An auxiliary method for extracting packet's IPv4/IPv6 source address as string
+	 */
+	std::string getSrcIPString(pcpp::Packet& packet)
+	{
+		if (packet.isPacketOfType(pcpp::IP))
+			return packet.getLayerOfType<pcpp::IPLayer>()->getSrcIPAddress().toString();
+		return "miscellaneous";
+	}
+
+	/**
+	 * An auxiliary method for extracting packet's IPv4/IPv6 dest address string
+	 */
+	std::string getDstIPString(pcpp::Packet& packet)
+	{
+		if (packet.isPacketOfType(pcpp::IP))
+			return packet.getLayerOfType<pcpp::IPLayer>()->getDstIPAddress().toString();
+		return "miscellaneous";
+	}
+
+	/**
+	 * An auxiliary method for replacing '.' and ':' in IPv4/IPv6 addresses with '-'
+	 */
+	std::string hyphenIP(std::string ipVal)
+	{
+		// for IPv4 - replace '.' with '-'
+		int loc = ipVal.find(".");
+		while (loc >= 0)
+		{
+			ipVal.replace(loc, 1, "-");
+			loc = ipVal.find(".");
+		}
+
+		// for IPv6 - replace ':' with '-'
+		loc = ipVal.find(":");
+		while (loc >= 0)
+		{
+			ipVal.replace(loc, 1, "-");
+			loc = ipVal.find(":");
+		}
+
+		return ipVal;
+	}
+
+	void updateStringStream(std::ostringstream & sstream, const std::string & srcIp, uint16_t srcPort, const std::string & dstIp, uint16_t dstPort)
+	{
+		sstream << hyphenIP(srcIp);
+		sstream << "_";
+		sstream << srcPort;
+		sstream << "-";
+		sstream << hyphenIP(dstIp);
+		sstream << "_";
+		sstream << dstPort;
+	}
+
+	/**
+	 * Re-implement Splitter's getFileName() method, this time with the IPs/Ports/protocol value
+	 */
+	std::string getFileName(pcpp::Packet& packet, const std::string &outputPcapBasePath, int fileNumber)
+	{
+		std::ostringstream sstream;
+
+		// if it's not a TCP or UDP packet, put it in file #0
+		if (!packet.isPacketOfType(pcpp::TCP) && !packet.isPacketOfType(pcpp::UDP))
+		{
+			return Splitter::getFileName(packet, outputPcapBasePath, fileNumber);
+		}
+
+		sstream << "connection-";
+
+		if (packet.isPacketOfType(pcpp::TCP))
+		{
+			// extract TCP layer
+			pcpp::TcpLayer* tcpLayer = packet.getLayerOfType<pcpp::TcpLayer>();
+			if (tcpLayer != NULL)
+			{
+				uint16_t srcPort = tcpLayer->getSrcPort();
+				uint16_t dstPort = tcpLayer->getDstPort();
+
+				sstream << "tcp_";
+
+				if ((tcpLayer->getTcpHeader()->synFlag == 1) && (tcpLayer->getTcpHeader()->ackFlag == 0)){
```

Comment:
Can you please fix the curly brackets to match the code convention?
```cpp
if (...)
{
	...
}
```

Instead of:
```cpp
if (...) {
	...
}
```

Reply:
done